clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -370,6 +370,13 @@ static bool initTargetOptions(DiagnosticsEngine &Diags,
     break;
   }
 
+  if (TargetOpts.DFPEncoding == clang::TargetOptions::BinaryEncoding::BID)
+    Options.DFPEncoding = llvm::BinaryEncoding::BID;
+  else if (TargetOpts.DFPEncoding == clang::TargetOptions::BinaryEncoding::DPD)
+    Options.DFPEncoding = llvm::BinaryEncoding::DPD;
+  else
+    Options.DFPEncoding = llvm::BinaryEncoding::None;
+
```

Comment:
Hmm, this suggests that the enumeration that is currently defined in `clang/include/clang/Basic/TargetInfo.h` should get moved to `llvm/include/llvm/Target/TargetOptions.h`. Or perhaps should be incorporated into `APFloatBase` in `llvm/include/llvm/ADT/APFloat.h`. What do you think?

Reply:
Exactly what I have done :-) Not sure about APFloat.h. I will take a look. For now it's residing here.